### DISCUSSION TITLE
Output Directory for `exportDatatables` Function in Gradle Plugin

### DISCUSSION THREAD

**Discussion by yusasak1** [openrewrite/rewrite#4603] [Q&A] [answered]
Hi,

I have questions about exportDatatables function when using Gradle plugin.

1. When setExportDatatables is set to `true`, the CSV files are output to `<repository top directory>/build/datatables`. I expected them to be output to `<gradle project top directory>/build/datatables`. Is there a reason why they must be output to the top directory of the repository?

2. In a multi-project setup with Gradle, the DataTables from the OpenRewrite tasks executed for subprojects are also output to `<repository top directory>/build/datatables`. I expected them to be output to the `build/datatables` directory of each subproject. Is there a reason why they are not output to each subproject's directory? (For example, is it for the same reason that files output with the `--profile` option in Gradle are output to the top directory of Gradle, regardless of the subproject?)

I'm using: 

- Gradle 8.10
- OpenRewrite 2.20.0
- OpenRewrite Gradle plugin 6.24.0

```terminal
<Repository Top Directory>
+---build
|   \---rewrite
|       \---datatables // Currennt export Directory
|
+---GradleTop
|   +---.gradle
|   +---app
|   |   +---build
|   |   |   +---classes
|   |   |   |   \---java
|   |   |   |       +---main
|   |   |   |       \---test
|   |   |   +---distributions
|   |   |   +---generated
|   |   |   +---libs
|   |   |   +---reports
|   |   |   |   +---rewrite
|   |   |   |   \---tests
|   |   |   +---rewrite
|   |   |       \---datatables // or should be here?
|   |   |   +---scripts
|   |   |   +---test-results
|   |   |   \---tmp
|   |   \---src
|   |
|   +---build
|   |   \---rewrite
|   |       \---datatables // should be here?
|   |
|   +---buildSrc
|   |   +---.gradle
|   |   +---build
|   |   \---src
|   |
|   +---gradle
|   |   \---wrapper
|   \---utilities
|       +---build
|       |   +---classes
|       |   |   \---java
|       |   |       \---main
|       |   +---generated
|       |   +---libs
|       |   \---tmp
|       \---src
|           +---main
|           |   +---java
|           |   \---resources
|           \---test
|               \---resources
+---otherDirA
\---otherDirB
```

Thank you.

**Comment by timtebeek** [comment, marked as answer]
hi @yusasak1 ; welcome to the project! Writing out data tables from the Gradle plugin was contributed in https://github.com/openrewrite/rewrite-gradle-plugin/pull/310 ; from what you describe the way we determine the output path could be improved to look at Gradle project build directories, as opposed to the git repository base directory. I'll convert your discussion into an issue on the plugin, thanks!
- https://github.com/openrewrite/rewrite-gradle-plugin/issues/341

**Reply by yusasak1** [reply]
Hi @timtebeek, 
Thank you for creating the issue ticket. I have sent a PR. Could you please review it?

https://github.com/openrewrite/rewrite-gradle-plugin/pull/342

**Comment by yusasak1** [comment]
I will close this discussion because the PR has been merged. Thanks.
